pjsip/include/pjsua-lib/pjsua_internal.h

```diff
@@ -118,6 +118,14 @@ typedef struct call_answer
 } call_answer;
 
 
+/* Generic states */
```

Comment:
> I'm not sure why this has to be a generic state while the use is only for trickle ICE.

There seem to be some specific-but-semantically-generic state enums already in the library, e.g: `pjsua_state, pj_ice_sess_checklist_state`, so I was thinking why introduce another specific one instead of a generic one, just in case it can be reusable by other feature in the future.

> And can this be put in pjnath instead so that it can be used by pjnath user?

Actually it is not part of trickle ICE itself, but part of the using protocol (i.e: specific for SIP), other using protocol may not need it.

> 
> After I said this, I suddenly remember, will it be possible to update `icedemo` as well to provide an example for pjnath user who wants to use trickle ICE directly without pjsip/pjsua?
> (if this is quite major, perhaps need to schedule it for the next release instead).

Yes, it is perhaps a major change for icedemo. Actually there is one 'sample' already in pjnath-test, instead of using SDP, it conveys new candidates to other UA via shared global var (btw, haven't checked it with recent modifications), I think it can also be used as sample reference.